workflow/controller/controller.go

```diff
@@ -83,13 +82,12 @@ type WorkflowController struct {
 	session               sqlbuilder.Database
 	offloadNodeStatusRepo sqldb.OffloadNodeStatusRepo
 	wfArchive             sqldb.WorkflowArchive
-	Metrics               map[string]common.Metric
+	metrics               metrics.Metrics
```

Comment:
Haven't needed to create this as an interface. However I can see myself needing to do so:

* if we decide to fully allow metric disabling, we would need one to create the null metrics object

* for testing